src/App.tsx

```diff
@@ -1,60 +1,153 @@
-import React from 'react';
+import {
+  FC,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
-import classNames from 'classnames';
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
+import cn from 'classnames';
+import { PostsList } from './components/PostsList/PostsList';
+import { PostDetails } from './components/PostDetails/PostDetails';
+import { UserSelector } from './components/UserSelector/UserSelector';
 import { Loader } from './components/Loader';
+import { getUsers } from './api/users';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getPosts } from './api/posts';
+
+export const App: FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [selectedPostId, setSelectedPostId] = useState(0);
+  const [noPosts, setNoPosts] = useState(false);
+  const [isError, setIsError] = useState(false);
```

Comment:
It's more comfortable to have an `errorMessage` state and save there a string error. If it's empty then there are no errors, if it has value then it can be used for any error.